flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/compact/KeyValueAccumulator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.compact;
+
+import org.apache.flink.table.storage.filestore.KeyValue;
+
+import java.io.Serializable;
+
+/** Accumulators to merge multiple {@link KeyValue}s. */
+public interface KeyValueAccumulator extends Serializable {
```

Comment:
I prefer something like `org.apache.flink.api.common.accumulators.Accumulator`, also clone is needed for multiple threads:
```
void reset();

void add(KeyValue);

KeyValue getValue();

@Override
Accumulator clone();
```